### ISSUE TITLE
Custom OpenIdAuthenticationMapper not working 3.5.x

### ISSUE THREAD

**Issue by Kushan2021** [micronaut-projects/micronaut-security#1043]
### Expected Behavior

Micronaut version 3.4.1

Custom class created by using implement `OpenIdAuthenticationMapper` and it calls `createAuthenticationResponse`.

        `@Singleton
       public class CustomOpenIdAuthenticationMapper implements OpenIdAuthenticationMapper {

        @Override
        public AuthenticationResponse createAuthenticationResponse(String providerName,  OpenIdTokenResponse 
        tokenResponse,OpenIdClaims openIdClaims, @Nullable State state) {   

	   HashMap<String, Object> attrs = new HashMap<>();
	   attrs.put(OpenIdAuthenticationMapper.OPENID_TOKEN_KEY, tokenResponse.getIdToken());
	   attrs.put(OauthAuthenticationMapper.ACCESS_TOKEN_KEY, tokenResponse.getAccessToken());
           attrs.put(OauthAuthenticationMapper.REFRESH_TOKEN_KEY, tokenResponse.getRefreshToken());
           attrs.put(OauthAuthenticationMapper.PROVIDER_KEY, providerName);
           attrs.put("email", openIdClaims.getEmail());
		
           return AuthenticationResponse.success(openIdClaims.getEmail(), attrs);	
      }
    }`

### Actual Behaviour

Micronaut version 3.5.0

Custome class `createAuthenticationResponse` method not triggered.

### Environment Information

- Ubuntu 20.04
- JDK 17

### Version

3.5.0

**Comment by sdelamo** [comment]
Do you have a sample app?

**Comment by sdelamo** [comment]
I think you need to add a `@Named("yourOAuth2ClientName")` qualifier.

**Comment by sdelamo** [comment]
Have you tried adding `@Replaces(DefaultOpenIdAuthenticationMapper.class)`if you want to replace the global mapper? If not, you will need to add a `@Named` qualifier. 

https://micronaut-projects.github.io/micronaut-security/latest/guide/#openid-user-details`

**Comment by Kushan2021** [comment]
Yes I already used `@Named` qualifier, but forgot to add on sample code.

This is my application.yml.

`micronaut:
    security:
        oauth2:
            enabled: true
            clients:
                keycloak: 
                    enabled: true
                    grant-type: password
                    client-id: <<my client id>> 
                    client-secret: <<my client secret>>  
                    openid:
                        issuer: <<my openid issuer>> 
                        configuration-path: /.well-known/openid-configuration
                        jwks-uri: << my jwks uri >>
                        token:
                              auth-method: client-secret-post`

**Comment by Kushan2021** [comment]
Application works fine after upgraded into version 3.5.0 and downgraded this dependancy.

		`<dependency>
			<groupId>io.micronaut.security</groupId>
			<artifactId>micronaut-security-oauth2</artifactId>
			<version>3.4.1</version>
			<scope>compile</scope>
		</dependency>`

**Comment by Kushan2021** [comment]
Once I downgrade in version 3.4.1  `io.micronaut.security.authentication.Authenticator`  variable `authenticationProviders` contains `OpenIdPasswordAuthenticationProvider`. 
But new version  `authenticationProviders` contains empty array.

**Comment by j1cs** [comment]
Probably I'm hijacking this issue but i think is related.
it is possible to use micronaut.security.authentication: bearer with https://micronaut-projects.github.io/micronaut-security/latest/guide/#oauth2-user-details?
when I run my project doesn't seems to reach my OauthAuthenticationMapper implementation.
i made an example. never passed through here (never print) https://github.com/j1cs/oauth-mapper-error/blob/master/maintainer/src/main/java/me/jics/KeycloakAuthenticationMapper.java#L29

**Comment by wancharle** [comment]
> Have you tried adding `@Replaces(DefaultOpenIdAuthenticationMapper.class)`if you want to replace the global mapper? If not, you will need to add a `@Named` qualifier.
> 
> [https://micronaut-projects.github.io/micronaut-security/latest/guide/#openid-user-details`](https://micronaut-projects.github.io/micronaut-security/latest/guide/#openid-user-details%60)

I have the same problem on latest version and tried 'replaces' approauch too and doesnt work... So i downgraded to 3.4.1 like @Kushan2021 did and works.

Now i have a problem with downgrade because my app use PKCE which only works on more recent version.

Therefore,  will help much if micronaut teams fix that.
